chia/util/db_wrapper.py

```diff
@@ -259,6 +259,20 @@ async def writer_maybe_transaction(self) -> AsyncIterator[aiosqlite.Connection]:
                 finally:
                     self._current_writer = None
 
+    @contextlib.asynccontextmanager
+    async def reader(self) -> AsyncIterator[aiosqlite.Connection]:
+        async with self.reader_no_transaction() as connection:
+            if connection.in_transaction:
+                yield connection
+            else:
+                await connection.execute("BEGIN DEFERRED;")
```

Comment:
Does `SAVEPOINT` cause `in_transaction` to be true? If not, that would be problematic since I don't believe `BEGIN` is allowed to nest `SAVEPOINT`. (my reading of [this](https://sqlite.org/lang_savepoint.html))